tests/contracts/root_chain/test_process_exits.py

```diff
@@ -150,15 +150,28 @@ def test_finalize_exits_tx_race(testlang):
 
     testlang.forward_timestamp(2 * WEEK + 1)
     testlang.finalize_exits(NULL_ADDRESS, utxo1.spend_id, 1)
-    snapshot = testlang.ethtester.chain.snapshot()
     with pytest.raises(TransactionFailed):
         testlang.finalize_exits(NULL_ADDRESS, utxo1.spend_id, 3, startgas=1000000)
-    gas1 = testlang.ethtester.chain.last_gas_used()
-    testlang.ethtester.chain.revert(snapshot)
+    short_circuit_gas = testlang.ethtester.chain.last_gas_used()
+    assert short_circuit_gas < 67291  # value from _tx_race_normal
+
+
+def test_finalize_exits_tx_race_normal(testlang):
+    utxo1 = testlang.create_utxo()
+    utxo2 = testlang.create_utxo()
+    utxo3 = testlang.create_utxo()
+    utxo4 = testlang.create_utxo()
+    testlang.start_standard_exit(utxo1.owner, utxo1.spend_id)
+    testlang.start_standard_exit(utxo2.owner, utxo2.spend_id)
+    testlang.start_standard_exit(utxo3.owner, utxo3.spend_id)
+    testlang.start_standard_exit(utxo4.owner, utxo4.spend_id)
+
+    testlang.forward_timestamp(2 * WEEK + 1)
+    testlang.finalize_exits(NULL_ADDRESS, utxo1.spend_id, 1)
+
     testlang.finalize_exits(NULL_ADDRESS, utxo2.spend_id, 3)
-    gas2 = testlang.ethtester.chain.last_gas_used()
-    assert gas1 < 4000
-    assert gas2 > 30000
+    three_exits_gas = testlang.ethtester.chain.last_gas_used()
```

Comment:
@paulperegud Doesn't `last_gas_used` only get the gas used from the last transaction as opposed to the last three exits?

Reply:
Those three exits are being done as single Ethereum transaction. See third parameter in finalize_exits call.